src/modules/navigator/navigator_main.cpp

```diff
@@ -392,7 +410,7 @@ void Navigator::run()
 							rep->current.loiter_pattern = position_setpoint_s::LOITER_TYPE_ORBIT;
 						}
 
-						rep->current.loiter_direction_counter_clockwise = curr->current.loiter_direction_counter_clockwise;
+						rep->current.loiter_direction_counter_clockwise = (PX4_ISFINITE(cmd.param4) ? (cmd.param4 >= 1.0f) : false);
```

Comment:
Do we really need this separate handling? In the `only_alt_change_requested` case we, as the name says, want to keep everything beside the altitude as it was set before, no?

Reply:
I added this so when a 'change-altitude-only' command  is sent (i.e. param3 and param4 omitted), we explicitly apply the vehicle’s default loiter radius and direction, exactly the behavior people expect today. Without this, the vehicle would continue using the previous radius/direction settings, which can be surprising.

Also, this makes the behavior match the MAVLink spec’s intent that omitted parameters revert to defaults.

That said, I don't actually care but I do think it's the right thing.

Reply:
@ryanjAA I still challenge that it's expected that if you send a "change altitude" through QGC the loiter direction changes (see screen-recording). 
Could QGC send a full GoTo setpoint again? Probably, but it's currently not doing it, thus this change changes the current behavior.

[Screencast from 30-04-25 17:48:25.webm](https://github.com/user-attachments/assets/acdbdce1-3ce8-426d-a007-ea3d8c5b0610)

Reply:
> Also, this makes the behavior match the MAVLink spec’s intent that omitted parameters revert to defaults.

But then the lat/long fields should also not being assumed but reset to e.g. the current vehicle position. Currently there it keeps the lat/long from the previous setpoint.